features/ui/ui-kit-button/button.module.scss

```diff
@@ -0,0 +1,199 @@
+// .button {
+//   cursor: pointer;
+
+//   /* remove default button styles */
+//   border: none;
+//   margin: 0;
+//   padding: 0;
+//   background: transparent;
+//   line-height: normal;
+//   -webkit-font-smoothing: inherit;
+//   -moz-osx-font-smoothing: inherit;
+//   appearance: none;
+
+//   &::-moz-focus-inner {
+//     border: 0;
+//     padding: 0;
+//   }
+// }
+
+@use "@styles/color";
+@use "@styles/font";
+@use "@styles/space";
+
+.button {
+  // Common styles for all buttons
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border: none;
+  border-radius: 8px;
+  text-decoration: none;
+  cursor: pointer;
+  outline: none;
+
+  // padding: 10px 16px;
+  font-weight: bold;
+  transition:
+    background-color 0.3s,
+    color 0.3s;
+
+  &.small {
+    font: font.$text-sm-medium;
+    padding: 8px 14px;
+  }
+
+  &.medium {
+    font: font.$text-sm-medium;
+    padding: 10px 16px;
+  }
+
+  &.large {
+    font: font.$text-md-medium;
+    padding: 10px 18px;
+  }
+
+  &.xlarge {
+    font: font.$text-md-medium;
+    padding: 12px 20px;
+  }
+
+  &.primary {
+    background-color: color.$primary-600;
+    color: #fff;
+
+    &:hover {
+      background-color: color.$primary-700;
+    }
+
+    &:focus {
+      box-shadow:
+        0 0 0 4px #f4ebff,
+        0 1px 2px 0 rgb(16 24 40 / 5%);
```

Comment:
This could be extracted to a global scss variable similar to the color or space values